sinonjs/nise
# update package lock

### Purpose

With `package-lock.json` v1, this is extremely out-of-date with 165 low vulnerabilities and 14 moderate vulnerabilities upon installation. All but 2 are mitigated with the re-install using `NPM v6.10`. I delayed going to NPM v7 for it updates the package-lock.json format to lockfile v2.  Although it should be considered since the latest version out of npm is v7.7.0.  

Note, I tried v7 and it will require another wipe and rebuild of `package-lock.json` since oddly when I installed the current lockfile (v1) using NPM@7.7.0, it causes the `node_modules/.bin/` folder to be incorrectly configured and command line modules are missing.  It is also important to check if dependabot will be able to detect & support lockfile v2.

Additionally, modified the run-scripts since it contained deprecated features.  As stated in npm install instructions, prepublish is deprecated as of `NPM@v5` and should be changed to prepare for build instructions. The browserify bundle is a build instruction rather than a prepublishOnly feature so changed the run-script to prepare.

### How To Verify
1. Check out this branch
2. Run `npm install` with npm@v6
3. Check that lib/index.js was bundled as desired via prepare run-script.  Review installation log for errors and also check the `npm audit` results included.
